### PR TITLE
Date Picker styling

### PR DESCRIPTION
**Pull request by tasandberg**
* Slightly reorganized how we were pulling in default styles (import from npm package)
* Brought the pikaday calendar styles in line with design seen on the datepicker issue #41 
* Add v-if to <slot> child of ButtonDropdown to trigger a rerender of children there.  I was at a bit of a loss of how to restore the datepicker state when the dropdown is closed then reopened.  I don't think this rerender is expensive enough to worry about, but I do wish there were a more efficient way.
* Added a bit of logic to EventDateFilters component to cleanly switch between a startDate and endDate calendar.  Pikaday's container/field binding was causing some glitchiness out of the box.

screen shot 2017-04-02 at 11 46 47 pm

TODO:
- [x] Use `em` instead of pixels where possible in styles
- [x] Test and adjust for mobile screens

### CODE REVIEW

**Comment by tasandberg** [comment]
Dang: somethings breaking but I need to sleep.
```
Module build failed: SyntaxError: 'import' and 'export' may only appear at the top level (57:0)

  55 | };
  56 | 
> 57 | export default {
     | ^
  58 |   name: 'event-date-filters',
  59 |   props: ['filters', 'showTitle'],
  60 |   data: () => {

 @ ./src/components/EventDateFilters.vue 3:2-113
 @ ./src/components/Header.js
 @ ./src/initialize.js
 @ multi (webpack)-dev-server/client?http://localhost:8080 webpack/hot/dev-server ./src/initialize.js
```

File: src/components/ButtonDropdown.vue [hunk 1]
```diff
@@ -10,7 +10,7 @@
       </span>
     </button>
     <div v-show="isOpen" class="button-dropdown-content" :style="dropdownStyle">
-      <slot></slot>
+      <slot v-if="isOpen"></slot>
```
Comment (stuartsan) [inline src/components/ButtonDropdown.vue]: I think `v-show` might preserve the state on open/close, IIRC that is just doing an inline `display: none` toggle type of thing.

File: src/components/EventDateFilters.vue [hunk 2]
```diff
@@ -1,62 +1,92 @@
...
+        />
+        <strong>to</strong>
+        <input
+          v-on:click="currentCalendar = 'endDate'"
+          ref="endDate"
+          type="text"
+          :value="endDate"
+        />
+      </div>
+      <div id='calendar-container'>
```
Comment (stuartsan) [inline src/components/EventDateFilters.vue]: Is it possible here to use classes for the styles instead of ids, and if components need to be referenced give them a `ref`?

For example, `<div class="calendar-container" ref="container">` can then be referenced in the JS as `this.$refs.container`.

File: src/components/EventDateFilters.vue [hunk 3]
```diff
@@ -66,16 +96,21 @@ export default {
...
       minDate: new Date(),
+      bound: false,
       onSelect: function() {
         const date = this.getMoment();
         component.endDate = date.format(inputFormat);
         component.pikadayStartDate.setMaxDate(date.toDate());
       }
     });
+
+    // Focus the startDate field by default
```
Comment (stuartsan) [inline src/components/EventDateFilters.vue]: Nice

File: src/assets/styles/pikaday_custom.scss [hunk 4]
```diff
@@ -0,0 +1,55 @@
+@import "node_modules/pikaday/scss/pikaday.scss";
+
+$pd-day-hover-color: lighten($pd-day-color, 8%);
+
+// @Overwritten styles
+#calendar-container {
+  height: 215px;
```
Comment (stuartsan) [inline src/assets/styles/pikaday_custom.scss]: If it's possible to use ems/percentages/etc vs fixed pixels for sizes I think that would be ideal, as the rest of this project is doing that and it is more flexible. BUT I see this is kind of just forking the pikday styles and making adjustments, I have no idea how well that'd all work here. For the height, no big deal, fixed pixel it. If it's easy to swap in ems or rems for margin/padding values though, that would be sweet. (There is also an `$offset` variable defined, that might be handy).

File: src/assets/styles/pikaday_custom.scss [hunk 5]
```diff
@@ -0,0 +1,55 @@
+@import "node_modules/pikaday/scss/pikaday.scss";
```
Comment (stuartsan) [inline src/assets/styles/pikaday_custom.scss:1]: Ohhh I missed this before, sweet.

File: src/components/EventDateFilters.vue [hunk 6]
```diff
@@ -1,62 +1,92 @@
...
 </template>
 
 <script>
 import moment from 'moment';
+import Pikaday from 'pikaday';
 
 // sigh
 window.moment = moment;
 
-import Pikaday from 'pikaday';
```
Comment (stuartsan) [inline src/components/EventDateFilters.vue]: I think the reason I imported it that way was so that Pikaday would know that moment was available....I read somewhere that that was needed. But maybe it waits to check for moment until actually creating a Pikaday object? Anyway, if it works like this, cool. Might even be able to drop the `window.moment`, that would be nice...

**Comment by stuartsan** [comment]
I had a couple comments, otherwise this all looks great! I'd like to test it out a teeny bit before merging but that compilation failure...I'm super confused about why it's failing, it looks like the "top level" to me :man_shrugging: .

One other q: does it look ok on small screens, where the datepicker layout is a bit different?

File: src/components/ButtonDropdown.vue (hunk 1 above)
Comment (tasandberg) [reply src/components/ButtonDropdown.vue]: I use `v-if` to intentionally cause a destroy and re-mounting of the child component wrapped by the `slot`.  Since we may not want to force that behavior on all cases of ButtonDrowdown, it just occurred to me I could maybe add an optional param to ButtonDropdown to allow us to specify show/hide or hard rerender on a per-use basis.

Thoughts?

File: src/components/EventDateFilters.vue (hunk 2 above)
Comment (tasandberg) [reply src/components/EventDateFilters.vue]: I'll see if that jives with the pikaday constructor 👍

File: src/components/ButtonDropdown.vue (hunk 1 above)
Comment (stuartsan) [reply src/components/ButtonDropdown.vue]: Ahhh I see, I think I got it backwards before -- so you _want_ to destroy/reset the date picker's state each time it is closed then re-opened, rather than preserve it, eh? Guessing this is to re-trigger the focus on mount? I think this is fine w/o parametrizing it, and overall re: the expensiveness to re-render, meh. 

But -- I think this is redundant with the enclosing div's `v-show="isOpen"` directive -- like ideally that one could just be modified to `v-if`, BUT poking around with that locally I just noticed that changing it accordingly leads to a new issue with the `.toolbar-item` styles which are incidentally working now, but are fragile and lead to a can of worms.

TLDR: IMO fine to either do it as-is, or just leave out the change to the button dropdown and I'll submit a PR separately which updates that along with the styling regression.

File: src/components/EventDateFilters.vue [hunk 7]
```diff
@@ -1,62 +1,92 @@
...
   },
 	mounted() {
     const component = this;
 
     this.pikadayStartDate = new Pikaday({
-      field: this.$refs.startDate,
+      field: component.$refs.startDate,
       format: displayFormat,
-      position: 'bottom',
+      bound: false,
```
Comment (stuartsan) [inline src/components/EventDateFilters.vue:98]: Based on the mobile design:

image

As well as the desktop one from the AI file, I think the idea for both cases might be for the date picker calendar to be initially not visible, then on input focus, it joins the party. For desktop I think it looks great as-is in this PR, with the calendar immediately showing up in the dropdown, but it might be easier to have it work out of the box for both mobile and desktop if it starts out not visible.....however you think it works best tho, +1.

**Comment by stuartsan** [comment]
Added a couple additional comments, I poked around with it locally and it looks very nice on desktop.

File: src/components/ButtonDropdown.vue (hunk 1 above)
Comment (stuartsan) [reply src/components/ButtonDropdown.vue]: Update: the toolbar needed some tweaking and so along with that I updated the dropdown button to use `v-if` instead of `v-show` and fixed aforementioned can of styling worms over here: https://github.com/aclu-people-power/map/pull/71 . If this leads to the same effect of what you're aiming for here, could probably omit the changes to the button dropdown in this PR -- LMK.

**Comment by tasandberg** [comment]
### Changes
* Calendar only shows when you click on the date inputs (mobile and desktop)
* Viewport meta tag added for mobile scaling
* Mobile view fixed to fit the view responsively
* As a user, I can see a slide-up animation when I open the filters so that my mind is blown! :P (no one asked for this, I think it looks nice but we can nuke it.  I intentionally put it in a separate commit.)
screen shot 2017-04-06 at 10 20 52 am
screen shot 2017-04-06 at 10 20 57 am
screen shot 2017-04-06 at 10 21 09 am

File: src/assets/styles/header.scss [hunk 8]
```diff
@@ -66,15 +66,22 @@
   }
 }
 
+@keyframes slideUp {
```
Comment (stuartsan) [inline src/assets/styles/header.scss:69]: I dig the animation, but one thing it changes is the design, where the logos are maintained above the filters:

image

There is some code that makes that work by calculating the right position and setting an inline style: https://github.com/aclu-people-power/map/blob/master/src/components/Header.js#L35

So I'm not sure. Maybe check with one of the design folks? If they are cool with the full screen style, that's nice cause the animation is sweet, and we could drop that garbagey code setting an inline style too :grin: (Also could wire up the animation to work with the inline style thing but I suspect that might get complicated)

File: src/assets/styles/pikaday_custom.scss [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+$pd-day-hover-color: lighten($pd-day-color, 8%);
+
+.calendar-container {
```
Comment (stuartsan) [inline src/assets/styles/pikaday_custom.scss:5]: Two teeny calendar styling comments: for the colors of selected days I think the design is using `$aclu-blue`, and the text inputs have left-aligned text

image

**Review by stuartsan — APPROVED** [review]
This looks great!

File: src/assets/styles/header.scss [hunk 10]
```diff
@@ -68,7 +68,7 @@
 
 @keyframes slideUp {
   0% { top: 105vh; }
-  100% { top: 0vh; }
+  100% { top: 4.5em; }
```
Comment (stuartsan) [inline src/assets/styles/header.scss]: (Setting it in a fixed way like this is fairly brittle because if the styles in the header / logo area change it could break this too -- this is why it was calculating the top position via JS. But this should be fine, seems to look good across browsers / different zoom levels / etc. and I doubt there are going to be major changes to those other styles....probably my previous solution was overengineered ;) )

File: src/components/Header.js [hunk 11]
```diff
@@ -32,19 +32,7 @@ export default function(store){
       toggleView() {
         store.commit('viewToggled');
       },
-      toggleFilterEvents() {
```
Comment (stuartsan) [inline src/components/Header.js:35]: There is some corresponding code in the template that can also be chopped, the attribute reading `filterEventsTop` (also that definition in this file https://github.com/aclu-people-power/map/pull/69/commits/f96573df19d020dd5aa6e0348d2d1859bc90df8c#diff-6f7a94b64ddda80f88c2971ba5b8b1a3L17)
